projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -445,6 +450,38 @@ private function load_subtitle( $task ) {
 		return '';
 	}
 
+	/**
+	 * Loads the extended content for a task.
+	 *
+	 * @param Task $task A task definition.
+	 * @return array|null The extended content for the task.
+	 */
+	private function load_extended_content( $task ) {
+		if ( ! isset( $task['extended_content'] ) ) {
+			return null;
+		}
+
+		$extended_content_context = array();
+
+		foreach ( $task['extended_content'] as $context => $extended_content ) {
+			$loaded_extended_context = array();
+
+			$actions = $this->load_value_from_callback( $extended_content, 'actions', null, $task['id'] );
+			if ( $actions ) {
+				$loaded_extended_context['actions'] = $actions;
+			}
+
+			$content = $this->load_value_from_callback( $extended_content, 'content', null, $task['id'] );
+			if ( $actions ) {
+				$loaded_extended_context['content'] = $content;
+			}
+
+			$extended_content_context[ $context ] = $loaded_extended_context;
+		}
+
+		return $extended_content_context;
```

Comment:
The structure of the loop here doesn't quite match the data structure for the returned data, and it feels like we're still unnecessarily splitting up the action and content from each other. It also feels like `$this->load_value_from_callback()` isn't ideal for this more complex pattern. I would much rather we tackled this via one callback, along with a loop structure more like the following:
```suggestion
		$extended_content = array();

		foreach ( $task['extended_content'] as $context_index => $extended_content_data ) {
			$content_by_context = array();

			if ( isset( $extended_content_callback ) && is_callable( $extended_content_callback ) ) {
				$extended_content_for_context = call_user_func( $extended_content_callback, $task['id'] );

				if (
					is_array( $extended_content_for_context )
					&& isset( $extended_content_for_context['contexts'] )
					&& is_array( $extended_content_for_context['contexts'] )
					// require either actions or content - we might want both, or only require content
					&& ( isset( $extended_content_for_context['actions'] ) || isset( $extended_content_for_context['content'] ) )
				) {
					$current_contexts = $extended_content_for_context['contexts'];
					$current_content_updated = false;
					$current_context_data = array(
						'actions' => array(),
						'content' => array(),
					);

					if ( isset( $extended_content_for_context['actions'] ) && is_array( $extended_content_for_context['actions'] ) && array() !== $extended_content_for_context['actions'] ) {
						$current_context_data['actions'] = $extended_content_for_context['actions'];
						$current_content_updated = true;
					}

					if ( isset( $extended_content_for_context['content'] ) && is_array( $extended_content_for_context['content'] ) && array() !== $extended_content_for_context['content'] ) {
						$current_context_data['content'] = $extended_content_for_context['content'];
						$current_content_updated = true;
					}

					if ( $current_content_updated ) {
						foreach ( $current_contexts as $current_context ) {
							$extended_content[ $current_context ] = $current_context_data;
						}
					}
				}
			}
		}

		return $extended_content_context;
```